modules/engine/src/model/model.ts

```diff
@@ -301,6 +301,7 @@ export class Model {
     this.shaderFactory.release(this.pipeline.vs);
     this.shaderFactory.release(this.pipeline.fs);
     this._uniformStore.destroy();
+    this._gpuGeometry?.destroy();
```

Comment:
Fine for now, but we kind of need a concept of managed resources: What if the GPUGeometry was not created by this Model but paased in by the applications? One could argue that in that case it should not be managed by the model. 

(The idea here was to move towards requiring applications to creating and use GPUGeometries instead of Geometries so that Model doesn't create buffers itself.)

What if we added a `managed` prop / field on all GPU Resources that can be set during creation and modified. And we can add a `destroyIfManaged()` method.

If `Model` creates a resource, it sets the `managed` flag to true. If an app wants to copy a resource, it can set the managed flag to false and take over management.